src/main/java/com/google/code/or/common/util/MySQLUtils.java

```diff
@@ -116,30 +127,47 @@ public static int nanosForFractionalValue(int value, int width) {
 	}
 
 	public static java.util.Date toDatetime2(long value, int fraction, int width) {
+		final long millis = getMillisFromDatetime2(value, fraction, width);
+		return new java.util.Date(millis + (nanosForFractionalValue(fraction, width) / 1000000));
+	}
+
+	public static long getMillisFromDatetime2(long value, int fraction, int width) {

```

Comment:
none of the `getMillis` functions are using their fraction or width params, just ditch them